packages/task/src/browser/quick-open-task.ts

```diff
@@ -234,6 +298,22 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
             filteredRecentTasks, filteredConfiguredTasks, filteredProvidedTasks
         };
     }
+
+    private getGroupedTasksByWorkspaceFolder(tasks: TaskConfiguration[]): Map<string | undefined, TaskConfiguration[]> {
+        const grouped = new Map<string | undefined, TaskConfiguration[]>();
+        for (const task of tasks) {
+            const folder = task._scope;
+            if (grouped.has(folder)) {
+                grouped.get(folder)!.push(task);
+            } else {
+                grouped.set(folder, [task]);
+            }
+        }
+        for (const taskConfigs of grouped.values()) {
+            taskConfigs.sort((t1, t2) => t1.label.localeCompare(t2.label));
```

Comment:
btw, for case when `tasks.json` contains invalid configuration without `label`
`Terminal => Configure Tasks` can not be open without any messages to inform user about the cause, only error in browser console is displayed

![configure_error](https://user-images.githubusercontent.com/5676062/68067101-f3ba7080-fd4a-11e9-99d4-971caa41035c.png)

You can use the following configuration to reproduce it:
```
   {
      "type": "shell",
      "command": "sleep 2 && echo test",
      "problemMatcher": []
    }
```

I think it can be fixed within https://github.com/eclipse-theia/theia/issues/6482

Reply:
thank you for pointing it out ! I am working on this bug as part of 6482 :)